<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Lengerich Lab </title> <meta name="author" content="Lengerich Lab"> <meta name="description" content="AI for healthcare. Context matters. "> <meta property="og:site_name" content="Lengerich Lab"> <meta property="og:type" content="website"> <meta property="og:title" content="Lengerich Lab | About"> <meta property="og:url" content="https://lengerichlab.github.io/"> <meta property="og:description" content="AI for healthcare. Context matters. "> <meta property="og:locale" content="en"> <meta name="twitter:card" content="summary"> <meta name="twitter:title" content="About"> <meta name="twitter:description" content="AI for healthcare. Context matters. "> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/uw-crest-color-web-digital.svg?1180ff756d7161ca4079b3ed433b5f34"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://lengerichlab.github.io/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">About <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">People </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/resources/">Resources </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">Teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link"><i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title"> Lengerich Lab </h1> <p class="desc"><a href="//stat.wisc.edu">Department of Statistics</a> @ <a href="//wisc.edu">University of Wisconsin-Madison</a>.</p> </header> <article> <div class="clearfix"> <p>Welcome to the Lengerich lab at the University of Wisconsin-Madison. We develop ML methods and AI systems to advance precision medicine.</p> <p>Quick links:</p> <ul> <li><a href="/people">Our People</a></li> <li><a href="/publications">Our Publications</a></li> <li><a href="/projects">Our Current Projects</a></li> <li><a href="https://docs.google.com/forms/d/e/1FAIpQLSctsfvSMoNk9YnCYSGWTYE2he-dR3lpD9QOmQaRtCvwH3X7Hg/viewform" class="important-link" rel="external nofollow noopener" target="_blank">Apply to join us</a></li> </ul> <form action="https://buttondown.com/api/emails/embed-subscribe/lengerichlab" method="post" target="popupwindow" onsubmit="window.open('https://buttondown.com/lengerichlab', 'popupwindow')" class="embeddable-buttondown-form"> <label for="bd-email">Subscribe to email updates (semesterly):</label> <input type="email" name="email" id="bd-email"> <input type="submit" value="Subscribe"> </form> <p><img src="/assets/img/madison.jpeg" alt="Madison, Wisconsin" width="100%"></p> </div> <h2> <a href="/news/" style="color: inherit">news</a> </h2> <div class="news"> <div class="table-responsive" style="max-height: 60vw"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Mar 18, 2025</th> <td> Prof. Ben Lengerich discusses AI’s role in <strong>personalized healthcare</strong> in a faculty Q&amp;A. <a href="https://stat.wisc.edu/2025/03/17/faculty-qa-ben-lengerich-on-the-power-of-ai-in-medicine/" rel="external nofollow noopener" target="_blank">Read more</a>. </td> </tr> <tr> <th scope="row" style="width: 20%">Jan 26, 2025</th> <td> Sid has been named a <a href="https://www.societyforscience.org/regeneron-sts/2025-finalists/" rel="external nofollow noopener" target="_blank">finalist in the Regeneron Science Talent Search</a>. Congratulations Sid! </td> </tr> <tr> <th scope="row" style="width: 20%">Dec 18, 2024</th> <td> NVIDIA has awarded us an Academic Grant. Thank you for the support! </td> </tr> <tr> <th scope="row" style="width: 20%">Nov 04, 2024</th> <td> Two findings papers accepted at <a href="https://ahli.cc/ml4h/" rel="external nofollow noopener" target="_blank">ML4H 2024</a>: <a href="https://openreview.net/forum?id=3OYjWzqqC1" rel="external nofollow noopener" target="_blank"><em>From One to Zero: RAG-IM Adapts Language Models for Interpretable Zero-Shot Clinical Predictions</em></a>; and <a href="https://arxiv.org/abs/2411.10645" rel="external nofollow noopener" target="_blank"><em>Sample-Specific Models of Treatment Effects Explain Heterogeneity in Tuberculosis</em></a>. Congratulations to Sazan, Ethan, and the team! </td> </tr> <tr> <th scope="row" style="width: 20%">Aug 29, 2024</th> <td> We are writing an open, collaborative review paper on context-adaptive inference. Contribute <a href="https://github.com/lengerichLab/context-review/" rel="external nofollow noopener" target="_blank">here</a>. </td> </tr> </table> </div> </div> <h2> <a href="/publications/" style="color: inherit">selected publications</a> </h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div id="ellington2024contextualized" class="col-sm-10"> <div class="title">Contextualized: Heterogeneous Modeling Toolbox</div> <div class="author"> <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb N. Ellington</a>, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, Wesley Lo, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Aaron Alvarez, Andrea Rubbi, Manolis Kellis, Eric P. Xing' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>Journal of Open Source Software</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://www.theoj.org/joss-papers/joss.06469/10.21105.joss.06469.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Heterogeneous and context-dependent systems are common in real-world processes, such as those in biology, medicine, finance, and the social sciences. However, learning accurate and interpretable models of these heterogeneous systems remains an unsolved problem. Most statistical modeling approaches make strict assumptions about data homogeneity, leading to inaccurate models, while more flexible approaches are often too complex to interpret directly. Fundamentally, existing modeling tools force users to choose between accuracy and interpretability. Recent work on Contextualized Machine Learning (Lengerich et al., 2023) has introduced a new paradigm for modeling heterogeneous and context-dependent systems, which uses contextual metadata to generate sample-specific models, providing context-specific model-based insights and representing data heterogeneity with context-dependent model parameters. Here, we present Contextualized, a SKLearn-style Python package for estimating and analyzing personalized context-dependent models based on Contextualized Machine Learning. Contextualized implements two reusable and extensible concepts: a context encoder which translates sample context or metadata into model parameters, and sample-specific model which is defined by the context-specific parameters. With the flexibility of context-dependent parameters, each context-specific model can be a simple model class, such as a linear or Gaussian model, providing direct model-based interpretability without sacrificing overall accuracy.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ellington2024contextualized</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{The Open Journal}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{9}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{97}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{6469}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ellington, Caleb N. and Lengerich, Benjamin J. and Lo, Wesley and Alvarez, Aaron and Rubbi, Andrea and Kellis, Manolis and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Contextualized: Heterogeneous Modeling Toolbox}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of Open Source Software}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2020purifying" class="col-sm-10"> <div class="title">Purifying Interaction Effects with the Functional ANOVA: An Efficient Algorithm for Recovering Identifiable Additive Models</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Ben Lengerich</a>, Sarah Tan, Chun-Hao Chang, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Giles Hooker, Rich Caruana' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>In Proceedings of the Twenty Third International Conference on Artificial Intelligence and Statistics (AISTATS)</em> , 26–28 aug 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/1911.04974" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="http://proceedings.mlr.press/v108/lengerich20a.html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="http://proceedings.mlr.press/v108/lengerich20a/lengerich20a.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Models which estimate main effects of individual variables alongside interaction effects have an identifiability challenge: effects can be freely moved between main effects and interaction effects without changing the model prediction. This is a critical problem for interpretability because it permits “contradictory" models to represent the same function. To solve this problem, we propose pure interaction effects: variance in the outcome which cannot be represented by any subset of features. This definition has an equivalence with the Functional ANOVA decomposition. To compute this decomposition, we present a fast, exact algorithm that transforms any piecewise-constant function (such as a tree-based model) into a purified, canonical representation. We apply this algorithm to Generalized Additive Models with interactions trained on several datasets and show large disparity, including contradictions, between the apparent and the purified effects. These results underscore the need to specify data distributions and ensure identifiability before interpreting model parameters.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2020purifying</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Purifying Interaction Effects with the Functional ANOVA: An Efficient Algorithm for Recovering Identifiable Additive Models}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Ben and Tan, Sarah and Chang, Chun-Hao and Hooker, Giles and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the Twenty Third International Conference on Artificial Intelligence and Statistics (AISTATS)}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{2402--2412}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">{Chiappa, Silvia and Calandra, Roberto}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{108}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{Proceedings of Machine Learning Research}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="s">{26--28 Aug}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{PMLR}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{AISTATS}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Generalized Additive Models, Theory}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2019learning" class="col-sm-10"> <div class="title">Learning Sample-Specific Models with Low-Rank Personalized Regression</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.bryonaragam.com/" rel="external nofollow noopener" target="_blank">Bryon Aragam</a>, and <a href="https://www.cs.cmu.edu/~epxing/" rel="external nofollow noopener" target="_blank">Eric P. Xing</a> </div> <div class="periodical"> <em>In Advances in Neural Information Processing Systems (NeurIPS)</em> , 26–28 aug 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/1910.06939" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://proceedings.neurips.cc/paper/2019/file/52d2752b150f9c35ccb6869cbf074e48-Paper.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p> Modern applications of machine learning (ML) deal with increasingly heterogeneous datasets comprised of data collected from overlapping latent subpopulations. As a result, traditional models trained over large datasets may fail to recognize highly predictive localized effects in favour of weakly predictive global patterns. This is a problem because localized effects are critical to developing individualized policies and treatment plans in applications ranging from precision medicine to advertising. To address this challenge, we propose to estimate sample-specific models that tailor inference and prediction at the individual level. In contrast to classical ML models that estimate a single, complex model (or only a few complex models), our approach produces a model personalized to each sample. These sample-specific models can be studied to understand subgroup dynamics that go beyond coarse-grained class labels. Crucially, our approach does not assume that relationships between samples (e.g. a similarity network) are known a priori. Instead, we use unmodeled covariates to learn a latent distance metric over the samples. We apply this approach to financial, biomedical, and electoral data as well as simulated data and show that sample-specific models provide fine-grained interpretations of complicated phenomena without sacrificing predictive accuracy compared to state-of-the-art models such as deep neural networks. </p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2019learning</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Learning Sample-Specific Models with Low-Rank Personalized Regression}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Aragam, Bryon and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Advances in Neural Information Processing Systems (NeurIPS)}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Advances in Neural Information Processing Systems (NeurIPS)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{NeurIPS}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Contextualized}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2018personalized" class="col-sm-10"> <div class="title">Personalized Regression Enables Sample-specific Pan-cancer Analysis</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.bryonaragam.com/" rel="external nofollow noopener" target="_blank">Bryon Aragam</a>, and <a href="https://www.cs.cmu.edu/~epxing/" rel="external nofollow noopener" target="_blank">Eric P Xing</a> </div> <div class="periodical"> <em>Bioinformatics</em>, 26–28 aug 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="http://dx.doi.org/10.1093/bioinformatics/bty250" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>In many applications, inter-sample heterogeneity is crucial to understanding the complex biological processes under study. For example, in genomic analysis of cancers, each patient in a cohort may have a different driver mutation, making it difficult or impossible to identify causal mutations from an averaged view of the entire cohort. Unfortunately, many traditional methods for genomic analysis seek to estimate a single model which is shared by all samples in a population, ignoring this inter-sample heterogeneity entirely. In order to better understand patient heterogeneity, it is necessary to develop practical, personalized statistical models. To uncover this inter-sample heterogeneity, we propose a novel regularizer for achieving patient-specific personalized estimation. This regularizer operates by learning two latent distance metrics—one between personalized parameters and one between clinical covariates—and attempting to match the induced distances as closely as possible. Crucially, we do not assume these distance metrics are already known. Instead, we allow the data to dictate the structure of these latent distance metrics. Finally, we apply our method to learn patient-specific, interpretable models for a pan-cancer gene expression dataset containing samples from more than 30 distinct cancer types and find strong evidence of personalization effects between cancer types as well as between individuals. Our analysis uncovers sample-specific aberrations that are overlooked by population-level methods, suggesting a promising new path for precision analysis of complex diseases such as cancer.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lengerich2018personalized</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Aragam, Bryon and Xing, Eric P}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Personalized Regression Enables Sample-specific Pan-cancer Analysis}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{34}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{13}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{i178-i186}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{ISMB}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/bioinformatics/bty250}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{http://dx.doi.org/10.1093/bioinformatics/bty250}</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{/oup/backfile/content_public/journal/bioinformatics/34/13/10.1093_bioinformatics_bty250/1/bty250.pdf}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Contextualized, Statistical Genetics, Genomics, Cancer}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="bordt2024data" class="col-sm-10"> <div class="title">Data Science with LLMs and Interpretable Models</div> <div class="author"> <a href="https://sbordt.github.io/" rel="external nofollow noopener" target="_blank">Sebastian Bordt</a>, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Ben Lengerich</a>, <a href="https://www.microsoft.com/en-us/research/people/hanori/" rel="external nofollow noopener" target="_blank">Harsha Nori</a>, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Rich Caruana' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>AAAI Explainable AI for Science</em>, 26–28 aug 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/2308.01157" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Recent years have seen important advances in the building of interpretable models, machine learning models that are designed to be easily understood by humans. In this work, we show that large language models (LLMs) are remarkably good at working with interpretable models, too. In particular, we show that LLMs can describe, interpret, and debug Generalized Additive Models (GAMs). Combining the flexibility of LLMs with the breadth of statistical patterns accurately described by GAMs enables dataset summarization, question answering, and model critique. LLMs can also improve the interaction between domain experts and interpretable models, and generate hypotheses about the underlying phenomenon. We release TalkToEBM as an open-source LLM-GAM interface.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bordt2024data</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bordt, Sebastian and Lengerich, Ben and Nori, Harsha and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Data Science with LLMs and Interpretable Models}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{AAAI Explainable AI for Science}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{AAAI XAI4Sci}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, LLMs}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Lengerich Lab. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?b7816bd189846d29eded8745f9c4cf77"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.min.js" integrity="sha256-rjmgmaB99riUNcdlrDtcAiwtLIojSxNyUFdl+Qh+rB4=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> <script type="module" src="/assets/js/search/ninja-keys.min.js?601a2d3465e2a52bec38b600518d5f70"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script>let theme=determineComputedTheme();const ninjaKeys=document.querySelector("ninja-keys");"dark"===theme?ninjaKeys.classList.add("dark"):ninjaKeys.classList.remove("dark");const openSearchModal=()=>{const e=document.querySelector(".navbar-collapse");e.classList.contains("show")&&e.classList.remove("show"),ninjaKeys.open()};</script> <script>const ninja=document.querySelector("ninja-keys");ninja.data=[{id:"nav-about",title:"About",section:"Navigation",handler:()=>{window.location.href="/"}},{id:"nav-people",title:"People",description:"Interested in joining? See our open positions here, and apply here.",section:"Navigation",handler:()=>{window.location.href="/people/"}},{id:"nav-publications",title:"Publications",description:"Lab publications by year. For organization by project, see projects.",section:"Navigation",handler:()=>{window.location.href="/publications/"}},{id:"nav-projects",title:"Projects",description:"Our active research projects.",section:"Navigation",handler:()=>{window.location.href="/projects/"}},{id:"nav-resources",title:"Resources",description:"We value reproducible and open science.",section:"Navigation",handler:()=>{window.location.href="/resources/"}},{id:"nav-teaching",title:"Teaching",description:"Course materials",section:"Navigation",handler:()=>{window.location.href="/teaching/"}},{id:"nav-blog",title:"Blog",description:"",section:"Navigation",handler:()=>{window.location.href="/blog/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-3",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 3)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-3/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-2",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 2)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-2/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-1",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 1)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-1/"}},{id:"post-thoughts-from-my-faculty-application-experience",title:"Thoughts from my Faculty Application Experience",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/thoughts-from-my-faculty-application-experience/"}},{id:"post-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"post-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"post-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"post-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"post-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"post-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"news-lengerich-lab-is-open-at-the-university-of-wisconsin-madison",title:"Lengerich lab is open at the University of Wisconsin-Madison!",description:"",section:"News"},{id:"news-we-are-writing-an-open-collaborative-review-paper-on-context-adaptive-inference-contribute-lt-a-href-quot-https-github-com-lengerichlab-context-review-quot-gt-here-lt-a-gt",title:"We are writing an open, collaborative review paper on context-adaptive inference. Contribute &lt;a href=&quot;https://github.com/lengerichLab/context-review/&quot;&gt;here&lt;/a&gt;.",description:"",section:"News"},{id:"news-two-findings-papers-accepted-at-lt-a-href-quot-https-ahli-cc-ml4h-quot-gt-ml4h-2024-lt-a-gt-lt-a-href-quot-https-openreview-net-forum-id-3oyjwzqqc1-quot-gt-lt-em-gt-from-one-to-zero-rag-im-adapts-language-models-for-interpretable-zero-shot-clinical-predictions-lt-em-gt-lt-a-gt-and-lt-a-href-quot-https-arxiv-org-abs-2411-10645-quot-gt-lt-em-gt-sample-specific-models-of-treatment-effects-explain-heterogeneity-in-tuberculosis-lt-em-gt-lt-a-gt-congratulations-to-sazan-ethan-and-the-team",title:"Two findings papers accepted at &lt;a href=&quot;https://ahli.cc/ml4h/&quot;&gt;ML4H 2024&lt;/a&gt;: &lt;a href=&quot;https://openreview.net/forum?id=3OYjWzqqC1&quot;&gt;&lt;em&gt;From One to Zero: RAG-IM Adapts Language Models for Interpretable Zero-Shot Clinical Predictions&lt;/em&gt;&lt;/a&gt;; and &lt;a href=&quot;https://arxiv.org/abs/2411.10645&quot;&gt;&lt;em&gt;Sample-Specific Models of Treatment Effects Explain Heterogeneity in Tuberculosis&lt;/em&gt;&lt;/a&gt;. Congratulations to Sazan, Ethan, and the team!",description:"",section:"News"},{id:"news-nvidia-has-awarded-us-an-academic-grant-thank-you-for-the-support",title:"NVIDIA has awarded us an Academic Grant. Thank you for the support!",description:"",section:"News"},{id:"news-sid-has-been-named-a-lt-a-href-quot-https-www-societyforscience-org-regeneron-sts-2025-finalists-quot-gt-finalist-in-the-regeneron-science-talent-search-lt-a-gt-congratulations-sid",title:"Sid has been named a &lt;a href=&quot;https://www.societyforscience.org/regeneron-sts/2025-finalists/&quot;&gt;finalist in the Regeneron Science Talent Search&lt;/a&gt;. Congratulations Sid!",description:"",section:"News"},{id:"news-prof-ben-lengerich-discusses-ai-s-role-in-lt-strong-gt-personalized-healthcare-lt-strong-gt-in-a-faculty-q-amp-amp-a-lt-a-href-quot-https-stat-wisc-edu-2025-03-17-faculty-qa-ben-lengerich-on-the-power-of-ai-in-medicine-quot-gt-read-more-lt-a-gt",title:"Prof. Ben Lengerich discusses AI\u2019s role in &lt;strong&gt;personalized healthcare&lt;/strong&gt; in a faculty Q&amp;amp;A. &lt;a href=&quot;https://stat.wisc.edu/2025/03/17/faculty-qa-ben-lengerich-on-the-power-of-ai-in-medicine/&quot;&gt;Read more&lt;/a&gt;.",description:"",section:"News"},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-3",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 3)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-3/"}},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-2",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 2)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-2/"}},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-1",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 1)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-1/"}},{id:"posts-thoughts-from-my-faculty-application-experience",title:"Thoughts from my Faculty Application Experience",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/thoughts-from-my-faculty-application-experience/"}},{id:"posts-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"posts-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"posts-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"posts-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"posts-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"posts-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"projects-context-adaptive-systems",title:"Context-adaptive systems",description:"Training models to reveal heterogeneity by automatically adapting to context",section:"Projects",handler:()=>{window.location.href="/projects/1_contextualized/"}},{id:"projects-foundation-models-as-context",title:"Foundation models as context",description:"Bridging foundation models and statistical inference",section:"Projects",handler:()=>{window.location.href="/projects/2_foundation_models/"}},{id:"projects-intelligible-ml",title:"Intelligible ML",description:"Building glass-box models that are both predictive and interpretable",section:"Projects",handler:()=>{window.location.href="/projects/3_interpretable/"}},{id:"projects-automated-data-science",title:"Automated data science",description:"Automating data science with foundation models",section:"Projects",handler:()=>{window.location.href="/projects/4_data_science/"}},{id:"projects-genomics",title:"Genomics",description:"Dissecting the biological bases of disease",section:"Projects",handler:()=>{window.location.href="/projects/5_genomics/"}},{id:"projects-clinical-ai-tools",title:"Clinical AI Tools",description:"Building tools for practicing clinicians",section:"Projects",handler:()=>{window.location.href="/projects/6_clinical_ai/"}},{id:"projects-real-world-evidence",title:"Real-world evidence",description:"Building AI systems to reveal surprises in observational EHRs",section:"Projects",handler:()=>{window.location.href="/projects/7_rwe/"}},{id:"projects-maternal-health",title:"Maternal health",description:"Applications to understand risk factors underlying maternal morbidity.",section:"Projects",handler:()=>{window.location.href="/projects/8_smm/"}},{id:"socials-email",title:"Send email",section:"Socials",handler:()=>{window.open("mailto:%4C%61%73%74%4E%61%6D%65@%77%69%73%63.%65%64%75","_blank")}},{id:"socials-google-scholar",title:"Google Scholar",section:"Socials",handler:()=>{window.open("https://scholar.google.com/citations?user=a1Ck1CMAAAAJ","_blank")}},{id:"socials-rss",title:"RSS Feed",section:"Socials",handler:()=>{window.open("/feed.xml","_blank")}},{id:"light-theme",title:"Change theme to light",description:"Change the theme of the site to Light",section:"Theme",handler:()=>{setThemeSetting("light")}},{id:"dark-theme",title:"Change theme to dark",description:"Change the theme of the site to Dark",section:"Theme",handler:()=>{setThemeSetting("dark")}},{id:"system-theme",title:"Use system default theme",description:"Change the theme of the site to System Default",section:"Theme",handler:()=>{setThemeSetting("system")}}];</script> </body> </html>