<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Publications | Lengerich Lab </title> <meta name="author" content="Lengerich Lab"> <meta name="description" content="Lab publications by year. For organization by project, see &lt;a href=/projects&gt;projects&lt;/a&gt;."> <meta property="og:site_name" content="Lengerich Lab"> <meta property="og:type" content="website"> <meta property="og:title" content="Lengerich Lab | Publications"> <meta property="og:url" content="https://lengerichlab.github.io/publications/"> <meta property="og:description" content="Lab publications by year. For organization by project, see &lt;a href=/projects&gt;projects&lt;/a&gt;."> <meta property="og:locale" content="en"> <meta name="twitter:card" content="summary"> <meta name="twitter:title" content="Publications"> <meta name="twitter:description" content="Lab publications by year. For organization by project, see &lt;a href=/projects&gt;projects&lt;/a&gt;."> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/uw-crest-color-web-digital.svg?1180ff756d7161ca4079b3ed433b5f34"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://lengerichlab.github.io/publications/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Lengerich Lab </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">About </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">People </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">Publications <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">Repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">Teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link"><i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Publications</h1> <p class="post-description">Lab publications by year. For organization by project, see <a href="/projects">projects</a>.</p> </header> <article> <div class="publications"> <h2 class="bibliography">2024</h2> <ol class="bibliography"><li> <div class="row"> <div id="ellington2024contextualized" class="col-sm-10"> <div class="title">Contextualized: Heterogeneous Modeling Toolbox</div> <div class="author"> <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb N. Ellington</a>, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, Wesley Lo, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Aaron Alvarez, Andrea Rubbi, Manolis Kellis, Eric P. Xing' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>Journal of Open Source Software</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://www.theoj.org/joss-papers/joss.06469/10.21105.joss.06469.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Heterogeneous and context-dependent systems are common in real-world processes, such as those in biology, medicine, finance, and the social sciences. However, learning accurate and interpretable models of these heterogeneous systems remains an unsolved problem. Most statistical modeling approaches make strict assumptions about data homogeneity, leading to inaccurate models, while more flexible approaches are often too complex to interpret directly. Fundamentally, existing modeling tools force users to choose between accuracy and interpretability. Recent work on Contextualized Machine Learning (Lengerich et al., 2023) has introduced a new paradigm for modeling heterogeneous and context-dependent systems, which uses contextual metadata to generate sample-specific models, providing context-specific model-based insights and representing data heterogeneity with context-dependent model parameters. Here, we present Contextualized, a SKLearn-style Python package for estimating and analyzing personalized context-dependent models based on Contextualized Machine Learning. Contextualized implements two reusable and extensible concepts: a context encoder which translates sample context or metadata into model parameters, and sample-specific model which is defined by the context-specific parameters. With the flexibility of context-dependent parameters, each context-specific model can be a simple model class, such as a linear or Gaussian model, providing direct model-based interpretability without sacrificing overall accuracy.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ellington2024contextualized</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{The Open Journal}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{9}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{97}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{6469}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ellington, Caleb N. and Lengerich, Benjamin J. and Lo, Wesley and Alvarez, Aaron and Rubbi, Andrea and Kellis, Manolis and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Contextualized: Heterogeneous Modeling Toolbox}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of Open Source Software}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div id="bosschieter2023interpretable" class="col-sm-10"> <div class="title">Interpretable Predictive Models to Understand Risk Factors for Maternal and Fetal Outcomes</div> <div class="author"> Tomas M. Bosschieter, Zifei Xu, Hui Lan, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Benjamin J. Lengerich, Harsha Nori, Ian Painter, Vivienne Souter, Rich Caruana' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>Journal of Healthcare Informatics Research</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://link.springer.com/article/10.1007/s41666-023-00151-4" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Although most pregnancies result in a good outcome, complications are not uncommon and can be associated with serious implications for mothers and babies. Predictive modeling has the potential to improve outcomes through a better understanding of risk factors, heightened surveillance for high-risk patients, and more timely and appropriate interventions, thereby helping obstetricians deliver better care. We identify and study the most important risk factors for four types of pregnancy complications: (i) severe maternal morbidity, (ii) shoulder dystocia, (iii) preterm preeclampsia, and (iv) antepartum stillbirth. We use an Explainable Boosting Machine (EBM), a high-accuracy glass-box learning method, for the prediction and identification of important risk factors. We undertake external validation and perform an extensive robustness analysis of the EBM models. EBMs match the accuracy of other black-box ML methods, such as deep neural networks and random forests, and outperform logistic regression, while being more interpretable. EBMs prove to be robust. The interpretability of the EBM models reveal surprising insights into the features contributing to risk (e.g., maternal height is the second most important feature for shoulder dystocia) and may have potential for clinical application in the prediction and prevention of serious complications in pregnancy.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bosschieter2023interpretable</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Interpretable Predictive Models to Understand Risk Factors for Maternal and Fetal Outcomes}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bosschieter, Tomas M. and Xu, Zifei and Lan, Hui and Lengerich, Benjamin J. and Nori, Harsha and Painter, Ian and Souter, Vivienne and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of Healthcare Informatics Research}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{JHIR}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="bordt2024data" class="col-sm-10"> <div class="title">Data Science with LLMs and Interpretable Models</div> <div class="author"> Sebastian Bordt, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Ben Lengerich</a>, Harsha Nori, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Rich Caruana' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>AAAI Explainable AI for Science</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Recent years have seen important advances in the building of interpretable models, machine learning models that are designed to be easily understood by humans. In this work, we show that large language models (LLMs) are remarkably good at working with interpretable models, too. In particular, we show that LLMs can describe, interpret, and debug Generalized Additive Models (GAMs). Combining the flexibility of LLMs with the breadth of statistical patterns accurately described by GAMs enables dataset summarization, question answering, and model critique. LLMs can also improve the interaction between domain experts and interpretable models, and generate hypotheses about the underlying phenomenon. We release TalkToEBM as an open-source LLM-GAM interface.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bordt2024data</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bordt, Sebastian and Lengerich, Ben and Nori, Harsha and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Data Science with LLMs and Interpretable Models}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{AAAI Explainable AI for Science}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{AAAI XAI4Sci}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, LLMs}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"> <li> <div class="row"> <div id="lengerich2022automated" class="col-sm-10"> <div class="title">Automated interpretable discovery of heterogeneous treatment effectiveness: A COVID-19 case study</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J Lengerich</a>, Mark E Nunnally, Yin Aphinyanaphongs, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Caleb Ellington, Rich Caruana' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Journal of biomedical informatics</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://www.medrxiv.org/content/10.1101/2021.10.30.21265430v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a href="https://www.sciencedirect.com/science/article/abs/pii/S1532046422001022" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Testing multiple treatments for heterogeneous (varying) effectiveness with respect to many underlying risk factors requires many pairwise tests; we would like to instead automatically discover and visualize patient archetypes and predictors of treatment effectiveness using multitask machine learning. In this paper, we present a method to estimate these heterogeneous treatment effects with an interpretable hierarchical framework that uses additive models to visualize expected treatment benefits as a function of patient factors (identifying personalized treatment benefits) and concurrent treatments (identifying combinatorial treatment benefits). This method achieves state-of-the-art predictive power for COVID-19 in-hospital mortality and interpretable identification of heterogeneous treatment benefits. We first validate this method on the large public MIMIC-IV dataset of ICU patients to test recovery of heterogeneous treatment effects. Next we apply this method to a proprietary dataset of over 3000 patients hospitalized for COVID-19, and find evidence of heterogeneous treatment effectiveness predicted largely by indicators of inflammation and thrombosis risk: patients with few indicators of thrombosis risk benefit most from treatments against inflammation, while patients with few indicators of inflammation risk benefit most from treatments against thrombosis. This approach provides an automated methodology to discover heterogeneous and individualized effectiveness of treatments.</p> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2022dropout" class="col-sm-10"> <div class="title">Dropout as a Regularizer of Interaction Effects</div> <div class="author"> <em>Benjamin Lengerich</em>, <a href="https://www.cs.cmu.edu/~epxing/" rel="external nofollow noopener" target="_blank">Eric P. Xing</a>, and <a href="https://www.microsoft.com/en-us/research/people/rcaruana/" rel="external nofollow noopener" target="_blank">Rich Caruana</a> </div> <div class="periodical"> <em>In Proceedings of the Twenty Fifth International Conference on Artificial Intelligence and Statistics</em> , 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/2007.00823" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://proceedings.mlr.press/v151/lengerich22a.html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>We examine Dropout through the perspective of interactions: effects that require multiple variables. Given N variables, there are N \choose k possible sets of k variables (N univariate effects, \mathcalO(N^2) pairwise interactions, \mathcalO(N^3) 3-way interactions); we can thus imagine that models with large representational capacity could be dominated by high-order interactions. In this paper, we show that Dropout contributes a regularization effect which helps neural networks (NNs) explore functions of lower-order interactions before considering functions of higher-order interactions. Dropout imposes this regularization by reducing the effective learning rate of higher-order interactions. As a result, Dropout encourages models to learn lower-order functions of additive components. This understanding of Dropout has implications for choosing Dropout rates: higher Dropout rates should be used when we need stronger regularization against interactions. This perspective also issues caution against using Dropout to measure term salience because Dropout regularizes against high-order interactions. Finally, this view of Dropout as a regularizer of interactions provides insight into the varying effectiveness of Dropout across architectures and datasets. We also compare Dropout to weight decay and early stopping and find that it is difficult to obtain the same regularization with these alternatives.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2022dropout</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Dropout as a Regularizer of Interaction Effects}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin and Xing, Eric P. and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{{Proceedings of the Twenty Fifth International Conference on Artificial Intelligence and Statistics}}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{AISTATS}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the Twenty Fifth International Conference on Artificial Intelligence and Statistics}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Deep Learning, Theory}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lee2022ten" class="col-sm-10"> <div class="title">Ten quick tips for deep learning in biology</div> <div class="author"> Benjamin D Lee, Anthony Gitter, Casey S Greene, and <span class="more-authors" title="click to view 17 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '17 more authors' ? 'Sebastian Raschka, Finlay Maguire, Alexander J Titus, Michael D Kessler, Alexandra J Lee, Marc G Chevrette, Paul Allen Stewart, Thiago Britto-Borges, Evan M. Cofer Cofer, Kun-Hsing Yu, Juan Jose Carmona, Elana J. Fertig, Alexandr A. Kalinin, Brandon Signal, Benjamin J. Lengerich, Timothy J. Jr. Triche, Simina M. Boca' : '17 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">17 more authors</span> </div> <div class="periodical"> <em>PLoS computational biology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009803" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lee2022ten</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Ten quick tips for deep learning in biology}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lee, Benjamin D and Gitter, Anthony and Greene, Casey S and Raschka, Sebastian and Maguire, Finlay and Titus, Alexander J and Kessler, Michael D and Lee, Alexandra J and Chevrette, Marc G and Stewart, Paul Allen and Britto-Borges, Thiago and Cofer, Evan M. Cofer and Yu, Kun-Hsing and Carmona, Juan Jose and Fertig, Elana J. and Kalinin, Alexandr A. and Signal, Brandon and Lengerich, Benjamin J. and Triche, Timothy J. Jr. and Boca, Simina M.}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{PLoS computational biology}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{PLoS CompBio}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{18}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{3}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{e1009803}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Public Library of Science San Francisco, CA USA}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Deep Learning, Biology, Computational Genomics}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="bosschieter2022smfm2" class="col-sm-10"> <div class="title">Unique insights into risk factors for antepartum stillbirth using explainable AI</div> <div class="author"> Tomas Bosschieter, Zifei Xu, Hui Lan, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Benjamin Lengerich, Harsha Nori, Kristin Sitcov, Ian Painter, Vivienne Souter, Rich Caruana' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bosschieter2022smfm2</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Unique insights into risk factors for antepartum stillbirth using explainable AI}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bosschieter, Tomas and Xu, Zifei and Lan, Hui and Lengerich, Benjamin and Nori, Harsha and Sitcov, Kristin and Painter, Ian and Souter, Vivienne and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{SMFM}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(22)01587-3/abstract}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lan2022smfm" class="col-sm-10"> <div class="title">Understanding risk factors for shoulder dystocia using interpretable machine learning</div> <div class="author"> Hui Lan, Zifei Xu, Tomas Bosschieter, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Benjamin Lengerich, Harsha Nori, Kristin Sitcov, Ian Painter, Vivienne Souter, Rich Caruana' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lan2022smfm</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Understanding risk factors for shoulder dystocia using interpretable machine learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lan, Hui and Xu, Zifei and Bosschieter, Tomas and Lengerich, Benjamin and Nori, Harsha and Sitcov, Kristin and Painter, Ian and Souter, Vivienne and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{SMFM}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(22)02139-1/abstract}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="bosschieter2022smfm" class="col-sm-10"> <div class="title">Preterm preeclampsia prediction using intelligible machine learning</div> <div class="author"> Tomas Bosschieter, Zifei Xu, Hui Lan, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Benjamin Lengerich, Harsha Nori, Kristin Sitcov, Ian Painter, Vivienne Souter, Rich Caruana' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bosschieter2022smfm</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Preterm preeclampsia prediction using intelligible machine learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bosschieter, Tomas and Xu, Zifei and Lan, Hui and Lengerich, Benjamin and Nori, Harsha and Sitcov, Kristin and Painter, Ian and Souter, Vivienne and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{SMFM}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(22)01596-4/abstract}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="xu2022smfm" class="col-sm-10"> <div class="title">Predicting severe maternal morbidity at admission for delivery using intelligible machine learning</div> <div class="author"> Zifei Xu, Tomas Bosschieter, Hui Lan, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Benjamin Lengerich, Harsha Nori, Kristin Sitcov, Ian Painter, Vivienne Souter, Rich Caruana' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">xu2022smfm</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Predicting severe maternal morbidity at admission for delivery using intelligible machine learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Xu, Zifei and Bosschieter, Tomas and Lan, Hui and Lengerich, Benjamin and Nori, Harsha and Sitcov, Kristin and Painter, Ian and Souter, Vivienne and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{SMFM}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(22)01588-5/abstract}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"> <li> <div class="row"> <div id="agarwal2022neural" class="col-sm-10"> <div class="title">Neural Additive Models: Interpretable Machine Learning with Neural Nets</div> <div class="author"> Rishabh Agarwal, Levi Melnick, Nicholas Frosst, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Xuezhou Zhang, Ben Lengerich, Rich Caruana, Geoffrey E Hinton' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>Advances in Neural Information Processing Systems</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/2004.13912" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://proceedings.neurips.cc/paper/2021/hash/251bd0442dfcc53b5a761e050f8022b8-Abstract.html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Deep neural networks (DNNs) are powerful black-box predictors that have achieved impressive performance on a wide variety of tasks. However, their accuracy comes at the cost of intelligibility: it is usually unclear how they make their decisions. This hinders their applicability to high stakes decision-making domains such as healthcare. We propose Neural Additive Models (NAMs) which combine some of the expressivity of DNNs with the inherent intelligibility of generalized additive models. NAMs learn a linear combination of neural networks that each attend to a single input feature. These networks are trained jointly and can learn arbitrarily complex relationships between their input feature and the output. Our experiments on regression and classification datasets show that NAMs are more accurate than widely used intelligible models such as logistic regression and shallow decision trees. They perform similarly to existing state-of-the-art generalized additive models in accuracy, but are more flexible because they are based on neural nets instead of boosted trees. To demonstrate this, we show how NAMs can be used for multitask learning on synthetic data and on the COMPAS recidivism data due to their composability, and demonstrate that the differentiability of NAMs allows them to train more complex interpretable models for COVID-19.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">agarwal2022neural</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Neural Additive Models: Interpretable Machine Learning with Neural Nets}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Agarwal, Rishabh and Melnick, Levi and Frosst, Nicholas and Zhang, Xuezhou and Lengerich, Ben and Caruana, Rich and Hinton, Geoffrey E}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Advances in Neural Information Processing Systems}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{34}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{4699--4711}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{NeurIPS}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Deep Learning, Generalized Additive Models}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="chang2021how" class="col-sm-10"> <div class="title">How Interpretable and Trustworthy are GAMs?</div> <div class="author"> Chun-Hao Chang, Sarah Tan, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Ben Lengerich</a>, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Anna Goldenberg, Rich Caruana' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>In Proceedings of the 27th ACM SIGKDD International Conference on Knowledge Discovery &amp; Data Mining</em> , 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/2006.06466" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://dl.acm.org/doi/abs/10.1145/3447548.3467453" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Generalized additive models (GAMs) have become a leading model class for interpretable machine learning. However, there are many algorithms for training GAMs, and these can learn different or even contradictory models, while being equally accurate. Which GAM should we trust? In this paper, we quantitatively and qualitatively investigate a variety of GAM algorithms on real and simulated datasets. We find that GAMs with high feature sparsity (only using a few variables to make predictions) can miss patterns in the data and be unfair to rare subpopulations. Our results suggest that inductive bias plays a crucial role in what interpretable models learn and that tree-based GAMs represent the best balance of sparsity, fidelity and accuracy and thus appear to be the most trustworthy GAM models.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">chang2021how</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{How Interpretable and Trustworthy are GAMs?}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Chang, Chun-Hao and Tan, Sarah and Lengerich, Ben and Goldenberg, Anna and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Proceedings of the 27th ACM SIGKDD International Conference on Knowledge Discovery \&amp; Data Mining}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 27th ACM SIGKDD International Conference on Knowledge Discovery \&amp; Data Mining}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{KDD}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Generalized Additive Models}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2021length" class="col-sm-10"> <div class="title">Length of labor and severe maternal morbidity in the NTSV population</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.microsoft.com/en-us/research/people/rcaruana/" rel="external nofollow noopener" target="_blank">Rich Caruana</a>, William B Weeks, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Ian Painter, Sydney Spencer, Kristin Sitcov, Colleen Daly, Vivienne Souter' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lengerich2021length</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Length of labor and severe maternal morbidity in the NTSV population}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Caruana, Rich and Weeks, William B and Painter, Ian and Spencer, Sydney and Sitcov, Kristin and Daly, Colleen and Souter, Vivienne}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{224}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{S33}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(20)31386-7/pdf}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2021insights" class="col-sm-10"> <div class="title">Insights into severe maternal morbidity in the NTSV population</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.microsoft.com/en-us/research/people/rcaruana/" rel="external nofollow noopener" target="_blank">Rich Caruana</a>, William B Weeks, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Ian Painter, Sydney Spencer, Kristin Sitcov, Colleen Daly, Vivienne Souter' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>American Journal of Obstetrics &amp; Gynecology</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lengerich2021insights</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Insights into severe maternal morbidity in the NTSV population}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Caruana, Rich and Weeks, William B and Painter, Ian and Spencer, Sydney and Sitcov, Kristin and Daly, Colleen and Souter, Vivienne}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Journal of Obstetrics \&amp; Gynecology}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{224}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{S629--S630}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Elsevier}</span><span class="p">,</span>
  <span class="na">url_paper</span> <span class="p">=</span> <span class="s">{https://www.ajog.org/article/S0002-9378(20)32418-2/fulltext}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Pregnancy}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="caruana2021data" class="col-sm-10"> <div class="title">Data-Driven Patterns in Protective Effects of Ibuprofen and Ketorolac on Hospitalized Covid-19 Patients</div> <div class="author"> <a href="https://www.microsoft.com/en-us/research/people/rcaruana/" rel="external nofollow noopener" target="_blank">Rich Caruana</a>, <em>Benjamin Lengerich</em>, and Yin Aphinyanaphongs </div> <div class="periodical"> <em>In </em> , 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>The impact of nonsteroidal anti-inflammatory drugs (NSAIDs) on patients with Covid-19 has been unclear. A major reason for this uncertainty is the confounding between treatments, patient comorbidities, and illness severity. Here, we perform an observational analysis of over 3000 patients hospitalized for Covid-19 in a New York hospital system to identify the relationship between in-patient treatment with Ibuprofen or Ketorolac and mortality. Our analysis finds evidence consitent with a protective effect for Ibuprofen and Ketorolac, with evidence stronger for a protective effect of Ketorolac than for a protective effect of Ibuprofen.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">caruana2021data</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Data-Driven Patterns in Protective Effects of Ibuprofen and Ketorolac on Hospitalized Covid-19 Patients}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Caruana, Rich and Lengerich, Benjamin and Aphinyanaphongs, Yin}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{American Medical Informatics Association (AMIA) Annual Symposium}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">url_preprint</span> <span class="p">=</span> <span class="s">{https://www.medrxiv.org/content/10.1101/2021.06.15.21258991v1}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Healthcare, Covid-19}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div id="lengerich2020purifying" class="col-sm-10"> <div class="title">Purifying Interaction Effects with the Functional ANOVA: An Efficient Algorithm for Recovering Identifiable Additive Models</div> <div class="author"> <em>Benjamin Lengerich</em>, Sarah Tan, Chun-Hao Chang, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Giles Hooker, Rich Caruana' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>In Proceedings of the Twenty Third International Conference on Artificial Intelligence and Statistics</em> , 26–28 aug 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/1911.04974" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="http://proceedings.mlr.press/v108/lengerich20a.html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="http://proceedings.mlr.press/v108/lengerich20a/lengerich20a.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Models which estimate main effects of individual variables alongside interaction effects have an identifiability challenge: effects can be freely moved between main effects and interaction effects without changing the model prediction. This is a critical problem for interpretability because it permits “contradictory" models to represent the same function. To solve this problem, we propose pure interaction effects: variance in the outcome which cannot be represented by any subset of features. This definition has an equivalence with the Functional ANOVA decomposition. To compute this decomposition, we present a fast, exact algorithm that transforms any piecewise-constant function (such as a tree-based model) into a purified, canonical representation. We apply this algorithm to Generalized Additive Models with interactions trained on several datasets and show large disparity, including contradictions, between the apparent and the purified effects. These results underscore the need to specify data distributions and ensure identifiability before interpreting model parameters.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2020purifying</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Purifying Interaction Effects with the Functional ANOVA: An Efficient Algorithm for Recovering Identifiable Additive Models}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin and Tan, Sarah and Chang, Chun-Hao and Hooker, Giles and Caruana, Rich}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the Twenty Third International Conference on Artificial Intelligence and Statistics}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{2402--2412}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">{Chiappa, Silvia and Calandra, Roberto}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{108}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{Proceedings of Machine Learning Research}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="s">{26--28 Aug}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{PMLR}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{AISTATS}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Generalized Additive Models, Theory}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"><li> <div class="row"> <div id="lengerich2019learning" class="col-sm-10"> <div class="title">Learning Sample-Specific Models with Low-Rank Personalized Regression</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.bryonaragam.com/" rel="external nofollow noopener" target="_blank">Bryon Aragam</a>, and <a href="https://www.cs.cmu.edu/~epxing/" rel="external nofollow noopener" target="_blank">Eric P. Xing</a> </div> <div class="periodical"> <em>In Advances in Neural Information Processing Systems (NeurIPS)</em> , 26–28 aug 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/1910.06939" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://proceedings.neurips.cc/paper/2019/file/52d2752b150f9c35ccb6869cbf074e48-Paper.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p> Modern applications of machine learning (ML) deal with increasingly heterogeneous datasets comprised of data collected from overlapping latent subpopulations. As a result, traditional models trained over large datasets may fail to recognize highly predictive localized effects in favour of weakly predictive global patterns. This is a problem because localized effects are critical to developing individualized policies and treatment plans in applications ranging from precision medicine to advertising. To address this challenge, we propose to estimate sample-specific models that tailor inference and prediction at the individual level. In contrast to classical ML models that estimate a single, complex model (or only a few complex models), our approach produces a model personalized to each sample. These sample-specific models can be studied to understand subgroup dynamics that go beyond coarse-grained class labels. Crucially, our approach does not assume that relationships between samples (e.g. a similarity network) are known a priori. Instead, we use unmodeled covariates to learn a latent distance metric over the samples. We apply this approach to financial, biomedical, and electoral data as well as simulated data and show that sample-specific models provide fine-grained interpretations of complicated phenomena without sacrificing predictive accuracy compared to state-of-the-art models such as deep neural networks. </p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2019learning</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Learning Sample-Specific Models with Low-Rank Personalized Regression}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Aragam, Bryon and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Advances in Neural Information Processing Systems (NeurIPS)}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Advances in Neural Information Processing Systems (NeurIPS)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{NeurIPS}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Contextualized}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="bibliography">2018</h2> <ol class="bibliography"> <li> <div class="row"> <div id="wang2018precision" class="col-sm-10"> <div class="title">Precision Lasso: Accounting for Correlations and Linear Dependencies in High-Dimensional Genomic Data</div> <div class="author"> Haohan Wang, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.bryonaragam.com/" rel="external nofollow noopener" target="_blank">Bryon Aragam</a>, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Eric P Xing' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Bioinformatics</em>, 26–28 aug 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://academic.oup.com/bioinformatics/article/35/7/1181/5089232?login=true" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Association studies to discover links between genetic markers and phenotypes are central to bioinformatics. Methods of regularized regression, such as variants of the Lasso, are popular for this task. Despite the good predictive performance of these methods in the average case, they suffer from unstable selections of correlated variables and inconsistent selections of linearly dependent variables. Unfortunately, as we demonstrate empirically, such problematic situations of correlated and linearly dependent variables often exist in genomic datasets and lead to under-performance of classical methods of variable selection. To address these challenges, we propose the Precision Lasso. Precision Lasso is a Lasso variant that promotes sparse variable selection by regularization governed by the covariance and inverse covariance matrices of explanatory variables. We illustrate its capacity for stable and consistent variable selection in simulated data with highly correlated and linearly dependent variables. We then demonstrate the effectiveness of the Precision Lasso to select meaningful variables from transcriptomic profiles of breast cancer patients. Our results indicate that in settings with correlated and linearly dependent variables, the Precision Lasso outperforms popular methods of variable selection such as the Lasso, the Elastic Net and Minimax Concave Penalty (MCP) regression.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">wang2018precision</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Precision Lasso: Accounting for Correlations and Linear Dependencies in High-Dimensional Genomic Data}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Wang, Haohan and Lengerich, Benjamin J. and Aragam, Bryon and Xing, Eric P}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{35}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{7}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1181--1187}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Oxford University Press}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Statistical Genetics, Genomics}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2018retrofitting" class="col-sm-10"> <div class="title">Retrofitting Distributional Embeddings to Knowledge Graphs with Functional Relations</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, Andrew Maas, and Christopher Potts </div> <div class="periodical"> <em>In Proceedings of the 27th International Conference on Computational Linguistics (COLING)</em> , 26–28 aug 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://arxiv.org/abs/1708.00112" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://aclanthology.org/C18-1205/" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Knowledge graphs are a versatile framework to encode richly structured data relationships, but it can be challenging to combine these graphs with unstructured data. Methods for retrofitting pre-trained entity representations to the structure of a knowledge graph typically assume that entities are embedded in a connected space and that relations imply similarity. However, useful knowledge graphs often contain diverse entities and relations (with potentially disjoint underlying corpora) which do not accord with these assumptions. To overcome these limitations, we present Functional Retrofitting, a framework that generalizes current retrofitting methods by explicitly modeling pairwise relations. Our framework can directly incorporate a variety of pairwise penalty functions previously developed for knowledge graph completion. Further, it allows users to encode, learn, and extract information about relation semantics. We present both linear and neural instantiations of the framework. Functional Retrofitting significantly outperforms existing retrofitting methods on complex knowledge graphs and loses no accuracy on simpler graphs (in which relations do imply similarity). Finally, we demonstrate the utility of the framework by predicting new drug–disease treatment pairs in a large, complex health knowledge graph.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lengerich2018retrofitting</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Retrofitting Distributional Embeddings to Knowledge Graphs with Functional Relations}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Maas, Andrew and Potts, Christopher}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Proceedings of the 27th International Conference on Computational Linguistics (COLING)}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 27th International Conference on Computational Linguistics (COLING)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{COLING}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Natural Language Processing, Knowledge Graphs}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="lengerich2018personalized" class="col-sm-10"> <div class="title">Personalized Regression Enables Sample-specific Pan-cancer Analysis</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, <a href="https://www.bryonaragam.com/" rel="external nofollow noopener" target="_blank">Bryon Aragam</a>, and <a href="https://www.cs.cmu.edu/~epxing/" rel="external nofollow noopener" target="_blank">Eric P Xing</a> </div> <div class="periodical"> <em>Bioinformatics</em>, 26–28 aug 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/https://www.biorxiv.org/content/early/2018/04/05/294496?" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="http://dx.doi.org/10.1093/bioinformatics/bty250" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>In many applications, inter-sample heterogeneity is crucial to understanding the complex biological processes under study. For example, in genomic analysis of cancers, each patient in a cohort may have a different driver mutation, making it difficult or impossible to identify causal mutations from an averaged view of the entire cohort. Unfortunately, many traditional methods for genomic analysis seek to estimate a single model which is shared by all samples in a population, ignoring this inter-sample heterogeneity entirely. In order to better understand patient heterogeneity, it is necessary to develop practical, personalized statistical models. To uncover this inter-sample heterogeneity, we propose a novel regularizer for achieving patient-specific personalized estimation. This regularizer operates by learning two latent distance metrics—one between personalized parameters and one between clinical covariates—and attempting to match the induced distances as closely as possible. Crucially, we do not assume these distance metrics are already known. Instead, we allow the data to dictate the structure of these latent distance metrics. Finally, we apply our method to learn patient-specific, interpretable models for a pan-cancer gene expression dataset containing samples from more than 30 distinct cancer types and find strong evidence of personalization effects between cancer types as well as between individuals. Our analysis uncovers sample-specific aberrations that are overlooked by population-level methods, suggesting a promising new path for precision analysis of complex diseases such as cancer.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lengerich2018personalized</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Benjamin J. and Aragam, Bryon and Xing, Eric P}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Personalized Regression Enables Sample-specific Pan-cancer Analysis}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{34}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{13}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{i178-i186}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{ISMB}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/bioinformatics/bty250}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{http://dx.doi.org/10.1093/bioinformatics/bty250}</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{/oup/backfile/content_public/journal/bioinformatics/34/13/10.1093_bioinformatics_bty250/1/bty250.pdf}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Interpretable, Contextualized, Statistical Genetics, Genomics, Cancer}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="ching2018opportunities" class="col-sm-10"> <div class="title">Opportunities and Obstacles for Deep Learning in Biology and Medicine</div> <div class="author"> Travers Ching, Daniel S. Himmelstein, Brett K. Beaulieu-Jones, and <span class="more-authors" title="click to view 33 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '33 more authors' ? 'Alexandr A. Kalinin, Brian T. Do, Gregory P. Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M. Hoffman, Wei Xie, Gail L. Rosen, Benjamin J. Lengerich, Johnny Israeli, Jack Lanchantin, Stephen Woloszynek, Anne E. Carpenter, Avanti Shrikumar, Jinbo Xu, Evan M. Cofer, Christopher A. Lavender, Srinivas C. Turaga, Amr M. Alexandari, Zhiyong Lu, David J. Harris, Dave DeCaprio, Yanjun Qi, Anshul Kundaje, Yifan Peng, Laura K. Wiley, Marwin H. S. Segler, Simina M. Boca, S. Joshua Swamidass, Austin Huang, Anthony Gitter, Casey S. Greene' : '33 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">33 more authors</span> </div> <div class="periodical"> <em>Journal of The Royal Society Interface</em>, 26–28 aug 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/http://www.biorxiv.org/content/biorxiv/early/2017/05/28/142760.full.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="http://rsif.royalsocietypublishing.org/content/15/141/20170387.full.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Deep learning describes a class of machine learning algorithms that are capable of combining raw inputs into layers of intermediate features. These algorithms have recently shown impressive results across a variety of domains. Biology and medicine are data-rich disciplines, but the data are complex and often ill-understood. Hence, deep learning techniques may be particularly well suited to solve problems of these fields. We examine applications of deep learning to a variety of biomedical problems—patient classification, fundamental biological processes and treatment of patients—and discuss whether deep learning will be able to transform these tasks or if the biomedical sphere poses unique challenges. Following from an extensive literature review, we find that deep learning has yet to revolutionize biomedicine or definitively resolve any of the most pressing challenges in the field, but promising advances have been made on the prior state of the art. Even though improvements over previous baselines have been modest in general, the recent progress indicates that deep learning methods will provide valuable means for speeding up or aiding human investigation. Though progress has been made linking a specific neural network’s prediction to input features, understanding how users should interpret these models to make testable hypotheses about the system under study remains an open challenge. Furthermore, the limited amount of labelled data for training presents problems in some domains, as do legal and privacy constraints on work with sensitive health records. Nonetheless, we foresee deep learning enabling changes at both bench and bedside with the potential to transform several areas of biology and medicine.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ching2018opportunities</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ching, Travers and Himmelstein, Daniel S. and Beaulieu-Jones, Brett K. and Kalinin, Alexandr A. and Do, Brian T. and Way, Gregory P. and Ferrero, Enrico and Agapow, Paul-Michael and Zietz, Michael and Hoffman, Michael M. and Xie, Wei and Rosen, Gail L. and Lengerich, Benjamin J. and Israeli, Johnny and Lanchantin, Jack and Woloszynek, Stephen and Carpenter, Anne E. and Shrikumar, Avanti and Xu, Jinbo and Cofer, Evan M. and Lavender, Christopher A. and Turaga, Srinivas C. and Alexandari, Amr M. and Lu, Zhiyong and Harris, David J. and DeCaprio, Dave and Qi, Yanjun and Kundaje, Anshul and Peng, Yifan and Wiley, Laura K. and Segler, Marwin H. S. and Boca, Simina M. and Swamidass, S. Joshua and Huang, Austin and Gitter, Anthony and Greene, Casey S.}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Opportunities and Obstacles for Deep Learning in Biology and Medicine}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{15}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{141}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1098/rsif.2017.0387}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{The Royal Society}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{JRSI}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{1742-5689}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{http://rsif.royalsocietypublishing.org/content/15/141/20170387}</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{http://rsif.royalsocietypublishing.org/content/15/141/20170387.full.pdf}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of The Royal Society Interface}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Deep Learning, Biology, Computational Genomics}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Lengerich Lab. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?b7816bd189846d29eded8745f9c4cf77"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.min.js" integrity="sha256-rjmgmaB99riUNcdlrDtcAiwtLIojSxNyUFdl+Qh+rB4=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> <script type="module" src="/assets/js/search/ninja-keys.min.js?601a2d3465e2a52bec38b600518d5f70"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script>let theme=determineComputedTheme();const ninjaKeys=document.querySelector("ninja-keys");"dark"===theme?ninjaKeys.classList.add("dark"):ninjaKeys.classList.remove("dark");const openSearchModal=()=>{const e=document.querySelector(".navbar-collapse");e.classList.contains("show")&&e.classList.remove("show"),ninjaKeys.open()};</script> <script>const ninja=document.querySelector("ninja-keys");ninja.data=[{id:"nav-about",title:"About",section:"Navigation",handler:()=>{window.location.href="/"}},{id:"nav-people",title:"People",description:"Interested in joining? Apply here.",section:"Navigation",handler:()=>{window.location.href="/people/"}},{id:"nav-publications",title:"Publications",description:"Lab publications by year. For organization by project, see projects.",section:"Navigation",handler:()=>{window.location.href="/publications/"}},{id:"nav-projects",title:"Projects",description:"Our active research projects.",section:"Navigation",handler:()=>{window.location.href="/projects/"}},{id:"nav-repositories",title:"Repositories",description:"We value reproducible and open science.",section:"Navigation",handler:()=>{window.location.href="/repositories/"}},{id:"nav-teaching",title:"Teaching",description:"Course materials",section:"Navigation",handler:()=>{window.location.href="/teaching/"}},{id:"nav-blog",title:"Blog",description:"",section:"Navigation",handler:()=>{window.location.href="/blog/"}},{id:"post-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"post-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"post-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"post-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"post-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"post-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"post-chatgpt-and-the-monty-hall-problem",title:"ChatGPT and the Monty Hall Problem",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/chatgpt-and-the-monty-hall-problem/"}},{id:"post-death-by-round-numbers-and-sharp-thresholds",title:"Death by Round Numbers and Sharp Thresholds",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/death-by-round-numbers-and-sharp-thresholds/"}},{id:"post-understanding-the-prevalence-of-sars-cov-2-with-limited-diagnostic-testing-capacity",title:"Understanding the Prevalence of SARS-CoV-2 with Limited Diagnostic Testing Capacity",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/understanding-the-prevalence-of-sars-cov-2-with-limited-diagnostic-testing-capacity/"}},{id:"post-a-model-for-each-sample-crazy",title:"A Model for Each Sample: Crazy?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-model-for-each-sample-crazy/"}},{id:"news-lengerich-lab-is-officially-starting-at-the-university-of-wisconsin-madison-apply-to-lt-a-href-quot-https-docs-google-com-document-d-1bb7wrfh0w7bg8biu7th-hlggawvx0f87jfhjf1saboo-edit-usp-sharing-quot-gt-join-us-lt-a-gt",title:"Lengerich lab is officially starting at the University of Wisconsin-Madison! Apply to &lt;a href=&quot;https://docs.google.com/document/d/1BB7WrFH0W7Bg8biU7tH_HLGGAWvx0f87jfhJf1SaBOo/edit?usp=sharing&quot;&gt;join us&lt;/a&gt;.",description:"",section:"News"},{id:"posts-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"posts-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"posts-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"posts-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"posts-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"posts-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"posts-chatgpt-and-the-monty-hall-problem",title:"ChatGPT and the Monty Hall Problem",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/chatgpt-and-the-monty-hall-problem/"}},{id:"posts-death-by-round-numbers-and-sharp-thresholds",title:"Death by Round Numbers and Sharp Thresholds",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/death-by-round-numbers-and-sharp-thresholds/"}},{id:"posts-understanding-the-prevalence-of-sars-cov-2-with-limited-diagnostic-testing-capacity",title:"Understanding the Prevalence of SARS-CoV-2 with Limited Diagnostic Testing Capacity",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/understanding-the-prevalence-of-sars-cov-2-with-limited-diagnostic-testing-capacity/"}},{id:"posts-a-model-for-each-sample-crazy",title:"A Model for Each Sample: Crazy?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-model-for-each-sample-crazy/"}},{id:"projects-contextualized-learning",title:"Contextualized learning",description:"Training ML models to reveal heterogeneity by automatically adapting to context",section:"Projects",handler:()=>{window.location.href="/projects/1_contextualized/"}},{id:"projects-foundational-inference",title:"Foundational inference",description:"Bridging foundation models and statistical inference",section:"Projects",handler:()=>{window.location.href="/projects/2_foundation_models/"}},{id:"projects-interpretable-ml",title:"Interpretable ML",description:"Building glass-box models that are both predictive and interpretable",section:"Projects",handler:()=>{window.location.href="/projects/3_interpretable/"}},{id:"projects-computational-genomics",title:"Computational genomics",description:"Dissecting the biological bases of disease",section:"Projects",handler:()=>{window.location.href="/projects/4_genomics/"}},{id:"projects-clinical-ai-tools",title:"Clinical AI Tools",description:"Building tools for practicing clinicians",section:"Projects",handler:()=>{window.location.href="/projects/5_clinical_ai/"}},{id:"projects-real-world-evidence",title:"Real-world evidence",description:"Building AI systems to reveal surprises in observational electronic health records",section:"Projects",handler:()=>{window.location.href="/projects/6_rwe/"}},{id:"socials-email",title:"Send email",section:"Socials",handler:()=>{window.open("mailto:%4C%61%73%74%4E%61%6D%65@%77%69%73%63.%65%64%75","_blank")}},{id:"socials-google-scholar",title:"Google Scholar",section:"Socials",handler:()=>{window.open("https://scholar.google.com/citations?user=a1Ck1CMAAAAJ","_blank")}},{id:"socials-rss",title:"RSS Feed",section:"Socials",handler:()=>{window.open("/feed.xml","_blank")}},{id:"light-theme",title:"Change theme to light",description:"Change the theme of the site to Light",section:"Theme",handler:()=>{setThemeSetting("light")}},{id:"dark-theme",title:"Change theme to dark",description:"Change the theme of the site to Dark",section:"Theme",handler:()=>{setThemeSetting("dark")}},{id:"system-theme",title:"Use system default theme",description:"Change the theme of the site to System Default",section:"Theme",handler:()=>{setThemeSetting("system")}}];</script> </body> </html>