<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Contextualized learning | Adaptive Inference Lab </title> <meta name="author" content=" "> <meta name="description" content="Models with parameters that adapt to patient, task, or environmental context to support personalized inference."> <meta property="og:site_name" content="Adaptive Inference Lab"> <meta property="og:type" content="website"> <meta property="og:title" content="Adaptive Inference Lab | Contextualized learning"> <meta property="og:url" content="https://adaptinfer.org/projects/1_contextualized/"> <meta property="og:description" content="Models with parameters that adapt to patient, task, or environmental context to support personalized inference."> <meta property="og:locale" content="en"> <meta name="twitter:card" content="summary"> <meta name="twitter:title" content="Contextualized learning"> <meta name="twitter:description" content="Models with parameters that adapt to patient, task, or environmental context to support personalized inference."> <meta name="twitter:site" content="@AdaptInferLab"> <meta name="twitter:creator" content="@AdaptInferLab"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/uw-crest-color-web-digital.svg?1180ff756d7161ca4079b3ed433b5f34"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://adaptinfer.org/projects/1_contextualized/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Adaptive Inference Lab </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/"> </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">People </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications </a> </li> <li class="nav-item active"> <a class="nav-link" href="/projects/">Projects <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/resources/">Resources </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">Teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link"><i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Contextualized learning</h1> <p class="post-description">Models with parameters that adapt to patient, task, or environmental context to support personalized inference.</p> </header> <article> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <img src="/assets/img/heterogeneity.jpeg" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="heterogeneity" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <p> Rather than using a single population model which operates identically for all contexts, context-adaptive systems estimate models which adapt to each local context. <a class="citation" href="#lengerich2023contextualized">(Lengerich et al., 2023)</a> </p> <p> We've built a Python package <a class="citation" href="#ellington2024contextualized">(Ellington et al., 2024)</a> and applied this paradigm to extend several biomedical tools: </p> <ul> <li>Indivdual treatment effects <a class="citation" href="#lengerich2022automated">(Lengerich et al., 2022)</a> </li> <li>Policy Recovery <a class="citation" href="#deuschel2024contextualized">(Deuschel et al., 2024)</a> </li> <li>Personalized Gene Regulatory Networks <a class="citation" href="#ellington2025learning">(Ellington et al., 2025)</a> </li> </ul> and we're continuing to build more tools with context-adaptive modeling. <p><br><br></p> </article> <h2>References</h2> <div class="publications"> <h2 class="bibliography">2025</h2> <ol class="bibliography"><li> <div class="row"> <div id="ellington2025learning" class="col-sm-10"> <div class="title">Learning to estimate sample-specific transcriptional networks for 7,000 tumors</div> <div class="author"> <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb N Ellington</a>, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J Lengerich</a>, Thomas BK Watkins, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Jiekun Yang, Abhinav K Adduri, Sazan Mahbub, Hanxi Xiao, Manolis Kellis, Eric P Xing' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>Proceedings of the National Academy of Sciences (PNAS)</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2411930122" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Cancers are shaped by somatic mutations, microenvironment, and patient background, each altering gene expression and regulation in complex ways, resulting in heterogeneous cellular states and dynamics. Inferring gene regulatory networks (GRNs) from expression data can help characterize this regulation-driven heterogeneity, but network inference requires many statistical samples, limiting GRNs to cluster-level analyses that ignore intracluster heterogeneity. We propose to move beyond coarse analyses of predefined subgroups by using contextualized learning, a multitask learning paradigm that uses multiview contexts including phenotypic, molecular, and environmental information to infer personalized models. With sample-specific contexts, contextualization enables sample-specific models and even generalizes at test time to predict network models for entirely unseen contexts. We unify three network model classes (Correlation, Markov, and Neighborhood Selection) and estimate context-specific GRNs for 7,997 tumors across 25 tumor types, using copy number and driver mutation profiles, tumor microenvironment, and patient demographics as model context. Our generative modeling approach allows us to predict GRNs for unseen tumor types based on a pan-cancer model of how somatic mutations affect gene regulation. Finally, contextualized networks enable GRN-based precision oncology by providing a structured view of expression dynamics at sample-specific resolution, explaining known biomarkers in terms of network-mediated effects and leading to subtypings that improve survival prognosis. We provide a SKLearn-style Python package https://contextualized.ml for learning and analyzing contextualized models, as well as interactive plotting tools for pan-cancer data exploration at https://github.com/cnellington/CancerContextualized</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ellington2025learning</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Learning to estimate sample-specific transcriptional networks for 7,000 tumors}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ellington, Caleb N and Lengerich, Benjamin J and Watkins, Thomas BK and Yang, Jiekun and Adduri, Abhinav K and Mahbub, Sazan and Xiao, Hanxi and Kellis, Manolis and Xing, Eric P}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Proceedings of the National Academy of Sciences (PNAS)}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{122}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{21}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{e2411930122}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{National Academy of Sciences}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="bibliography">2024</h2> <ol class="bibliography"> <li> <div class="row"> <div id="ellington2024contextualized" class="col-sm-10"> <div class="title">Contextualized: Heterogeneous Modeling Toolbox</div> <div class="author"> <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb N. Ellington</a>, <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J. Lengerich</a>, Wesley Lo, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Aaron Alvarez, Andrea Rubbi, Manolis Kellis, Eric P. Xing' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>Journal of Open Source Software</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://www.theoj.org/joss-papers/joss.06469/10.21105.joss.06469.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Heterogeneous and context-dependent systems are common in real-world processes, such as those in biology, medicine, finance, and the social sciences. However, learning accurate and interpretable models of these heterogeneous systems remains an unsolved problem. Most statistical modeling approaches make strict assumptions about data homogeneity, leading to inaccurate models, while more flexible approaches are often too complex to interpret directly. Fundamentally, existing modeling tools force users to choose between accuracy and interpretability. Recent work on Contextualized Machine Learning (Lengerich et al., 2023) has introduced a new paradigm for modeling heterogeneous and context-dependent systems, which uses contextual metadata to generate sample-specific models, providing context-specific model-based insights and representing data heterogeneity with context-dependent model parameters. Here, we present Contextualized, a SKLearn-style Python package for estimating and analyzing personalized context-dependent models based on Contextualized Machine Learning. Contextualized implements two reusable and extensible concepts: a context encoder which translates sample context or metadata into model parameters, and sample-specific model which is defined by the context-specific parameters. With the flexibility of context-dependent parameters, each context-specific model can be a simple model class, such as a linear or Gaussian model, providing direct model-based interpretability without sacrificing overall accuracy.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ellington2024contextualized</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.21105/joss.06469}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{The Open Journal}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{9}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{97}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{6469}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ellington, Caleb N. and Lengerich, Benjamin J. and Lo, Wesley and Alvarez, Aaron and Rubbi, Andrea and Kellis, Manolis and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Contextualized: Heterogeneous Modeling Toolbox}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of Open Source Software}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div id="deuschel2024contextualized" class="col-sm-10"> <div class="title">Contextualized Policy Recovery: Modeling and Interpreting Medical Decisions with Adaptive Imitation Learning</div> <div class="author"> <a href="https://scholar.google.com/citations?user=eGUxrfQAAAAJ" rel="external nofollow noopener" target="_blank">Jannik Deuschel</a>, <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb Ellington</a>, <a href="https://yingtaoluo.github.io/" rel="external nofollow noopener" target="_blank">Yingtao Luo</a>, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Ben Lengerich, Pascal Friederich, Eric Xing' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>International Conference on Machine Learning (ICML)</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/2310.07918" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://proceedings.mlr.press/v235/deuschel24a.html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Interpretable policy learning seeks to estimate intelligible decision policies from observed actions; however, existing models fall short by forcing a tradeoff between accuracy and interpretability. This tradeoff limits data-driven interpretations of human decision-making process. e.g. to audit medical decisions for biases and suboptimal practices, we require models of decision processes which provide concise descriptions of complex behaviors. Fundamentally, existing approaches are burdened by this tradeoff because they represent the underlying decision process as a universal policy, when in fact human decisions are dynamic and can change drastically with contextual information. Thus, we propose Contextualized Policy Recovery (CPR), which re-frames the problem of modeling complex decision processes as a multi-task learning problem in which complex decision policies are comprised of context-specific policies. CPR models each context-specific policy as a linear observation-to-action mapping, and generates new decision models on-demand as contexts are updated with new observations. CPR is compatible with fully offline and partially observable decision environments, and can be tailored to incorporate any recurrent black-box model or interpretable decision model. We assess CPR through studies on simulated and real data, achieving state-of-the-art performance on the canonical tasks of predicting antibiotic prescription in intensive care units (+22% AUROC vs. previous SOTA) and predicting MRI prescription for Alzheimer’s patients (+7.7% AUROC vs. previous SOTA). With this improvement in predictive performance, CPR closes the accuracy gap between interpretable and black-box methods for policy learning, allowing high-resolution exploration and analysis of context-specific decision models.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">deuschel2024contextualized</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Contextualized Policy Recovery: Modeling and Interpreting Medical Decisions with Adaptive Imitation Learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Deuschel, Jannik and Ellington, Caleb and Luo, Yingtao and Lengerich, Ben and Friederich, Pascal and Xing, Eric}</span><span class="p">,</span>
  <span class="na">informal_venue</span> <span class="p">=</span> <span class="s">{ICML}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{International Conference on Machine Learning (ICML)}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li> <div class="row"> <div id="lengerich2023contextualized" class="col-sm-10"> <div class="title">Contextualized Machine Learning</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Ben Lengerich</a>, <a href="https://cnellington.github.io/" rel="external nofollow noopener" target="_blank">Caleb N. Ellington</a>, <a href="https://andrearubbi.com/" rel="external nofollow noopener" target="_blank">Andrea Rubbi</a>, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Manolis Kellis, Eric P. Xing' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em></em> 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/2310.11340" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p> We examine Contextualized Machine Learning (ML), a paradigm for learning heterogeneous and context-dependent effects. Contextualized ML estimates heterogeneous functions by applying deep learning to the meta-relationship between contextual information and context-specific parametric models. This is a form of varying-coefficient modeling that unifies existing frameworks including cluster analysis and cohort modeling by introducing two reusable concepts: a context encoder which translates sample context into model parameters, and sample-specific model which operates on sample predictors. We review the process of developing contextualized models, nonparametric inference from contextualized models, and identifiability conditions of contextualized models. Finally, we present the open-source PyTorch package ContextualizedML.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">lengerich2023contextualized</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Contextualized Machine Learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lengerich, Ben and Ellington, Caleb N. and Rubbi, Andrea and Kellis, Manolis and Xing, Eric P.}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">archiveprefix</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"><li> <div class="row"> <div id="lengerich2022automated" class="col-sm-10"> <div class="title">Automated interpretable discovery of heterogeneous treatment effectiveness: A COVID-19 case study</div> <div class="author"> <a href="https://web.mit.edu/~blengeri/www/" rel="external nofollow noopener" target="_blank">Benjamin J Lengerich</a>, Mark E Nunnally, Yin Aphinyanaphongs, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Caleb Ellington, Rich Caruana' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Journal of biomedical informatics</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.medrxiv.org/content/10.1101/2021.10.30.21265430v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://www.sciencedirect.com/science/article/abs/pii/S1532046422001022" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="abstract hidden"> <p>Testing multiple treatments for heterogeneous (varying) effectiveness with respect to many underlying risk factors requires many pairwise tests; we would like to instead automatically discover and visualize patient archetypes and predictors of treatment effectiveness using multitask machine learning. In this paper, we present a method to estimate these heterogeneous treatment effects with an interpretable hierarchical framework that uses additive models to visualize expected treatment benefits as a function of patient factors (identifying personalized treatment benefits) and concurrent treatments (identifying combinatorial treatment benefits). This method achieves state-of-the-art predictive power for COVID-19 in-hospital mortality and interpretable identification of heterogeneous treatment benefits. We first validate this method on the large public MIMIC-IV dataset of ICU patients to test recovery of heterogeneous treatment effects. Next we apply this method to a proprietary dataset of over 3000 patients hospitalized for COVID-19, and find evidence of heterogeneous treatment effectiveness predicted largely by indicators of inflammation and thrombosis risk: patients with few indicators of thrombosis risk benefit most from treatments against inflammation, while patients with few indicators of inflammation risk benefit most from treatments against thrombosis. This approach provides an automated methodology to discover heterogeneous and individualized effectiveness of treatments.</p> </div> </div> </div> </li></ol> </div> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 . </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?b7816bd189846d29eded8745f9c4cf77"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.min.js" integrity="sha256-rjmgmaB99riUNcdlrDtcAiwtLIojSxNyUFdl+Qh+rB4=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> <script type="module" src="/assets/js/search/ninja-keys.min.js?601a2d3465e2a52bec38b600518d5f70"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script>let theme=determineComputedTheme();const ninjaKeys=document.querySelector("ninja-keys");"dark"===theme?ninjaKeys.classList.add("dark"):ninjaKeys.classList.remove("dark");const openSearchModal=()=>{const e=document.querySelector(".navbar-collapse");e.classList.contains("show")&&e.classList.remove("show"),ninjaKeys.open()};</script> <script>const ninja=document.querySelector("ninja-keys");ninja.data=[{id:"nav-",title:"",section:"Navigation",handler:()=>{window.location.href="/"}},{id:"nav-people",title:"People",description:"Interested in joining? Apply here.",section:"Navigation",handler:()=>{window.location.href="/people/"}},{id:"nav-publications",title:"Publications",description:"Lab publications by year. For organization by project, see projects.",section:"Navigation",handler:()=>{window.location.href="/publications/"}},{id:"nav-projects",title:"Projects",description:"Our active research projects.",section:"Navigation",handler:()=>{window.location.href="/projects/"}},{id:"nav-resources",title:"Resources",description:"We value reproducible and open science.",section:"Navigation",handler:()=>{window.location.href="/resources/"}},{id:"nav-teaching",title:"Teaching",description:"Course materials",section:"Navigation",handler:()=>{window.location.href="/teaching/"}},{id:"nav-blog",title:"Blog",description:"",section:"Navigation",handler:()=>{window.location.href="/blog/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-3",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 3)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-3/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-2",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 2)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-2/"}},{id:"post-a-patient-s-perspective-on-the-future-of-medical-ai-part-1",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 1)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-1/"}},{id:"post-thoughts-from-my-faculty-application-experience",title:"Thoughts from my Faculty Application Experience",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/thoughts-from-my-faculty-application-experience/"}},{id:"post-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"post-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"post-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"post-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"post-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"post-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"news-lengerich-lab-is-open-at-the-university-of-wisconsin-madison",title:"Lengerich lab is open at the University of Wisconsin-Madison!",description:"",section:"News"},{id:"news-we-are-writing-an-open-collaborative-review-paper-on-context-adaptive-inference-contribute-lt-a-href-quot-https-github-com-lengerichlab-context-review-quot-gt-here-lt-a-gt",title:"We are writing an open, collaborative review paper on context-adaptive inference. Contribute &lt;a href=&quot;https://github.com/lengerichLab/context-review/&quot;&gt;here&lt;/a&gt;.",description:"",section:"News"},{id:"news-two-findings-papers-accepted-at-lt-a-href-quot-https-ahli-cc-ml4h-quot-gt-ml4h-2024-lt-a-gt-lt-a-href-quot-https-openreview-net-forum-id-3oyjwzqqc1-quot-gt-lt-em-gt-from-one-to-zero-rag-im-adapts-language-models-for-interpretable-zero-shot-clinical-predictions-lt-em-gt-lt-a-gt-and-lt-a-href-quot-https-arxiv-org-abs-2411-10645-quot-gt-lt-em-gt-sample-specific-models-of-treatment-effects-explain-heterogeneity-in-tuberculosis-lt-em-gt-lt-a-gt-congratulations-to-sazan-ethan-and-the-team",title:"Two findings papers accepted at &lt;a href=&quot;https://ahli.cc/ml4h/&quot;&gt;ML4H 2024&lt;/a&gt;: &lt;a href=&quot;https://openreview.net/forum?id=3OYjWzqqC1&quot;&gt;&lt;em&gt;From One to Zero: RAG-IM Adapts Language Models for Interpretable Zero-Shot Clinical Predictions&lt;/em&gt;&lt;/a&gt;; and &lt;a href=&quot;https://arxiv.org/abs/2411.10645&quot;&gt;&lt;em&gt;Sample-Specific Models of Treatment Effects Explain Heterogeneity in Tuberculosis&lt;/em&gt;&lt;/a&gt;. Congratulations to Sazan, Ethan, and the team!",description:"",section:"News"},{id:"news-nvidia-has-awarded-us-an-academic-grant-thank-you-for-the-support",title:"NVIDIA has awarded us an Academic Grant. Thank you for the support!",description:"",section:"News"},{id:"news-sid-has-been-named-a-lt-a-href-quot-https-www-societyforscience-org-regeneron-sts-2025-finalists-quot-gt-finalist-in-the-regeneron-science-talent-search-lt-a-gt-congratulations-sid",title:"Sid has been named a &lt;a href=&quot;https://www.societyforscience.org/regeneron-sts/2025-finalists/&quot;&gt;finalist in the Regeneron Science Talent Search&lt;/a&gt;. Congratulations Sid!",description:"",section:"News"},{id:"news-prof-ben-lengerich-discusses-ai-s-role-in-lt-strong-gt-personalized-healthcare-lt-strong-gt-in-a-faculty-q-amp-amp-a-lt-a-href-quot-https-stat-wisc-edu-2025-03-17-faculty-qa-ben-lengerich-on-the-power-of-ai-in-medicine-quot-gt-read-more-lt-a-gt",title:"Prof. Ben Lengerich discusses AI\u2019s role in &lt;strong&gt;personalized healthcare&lt;/strong&gt; in a faculty Q&amp;amp;A. &lt;a href=&quot;https://stat.wisc.edu/2025/03/17/faculty-qa-ben-lengerich-on-the-power-of-ai-in-medicine/&quot;&gt;Read more&lt;/a&gt;.",description:"",section:"News"},{id:"news-our-paper-lt-a-href-quot-https-arxiv-org-abs-2505-20353-quot-gt-fastcache-lt-a-gt-is-published-in-lt-em-gt-cvpr-base-lt-em-gt-congrats-to-first-author-dong",title:"Our paper &lt;a href=&quot;https://arxiv.org/abs/2505.20353&quot;&gt;FastCache&lt;/a&gt; is published in &lt;em&gt;CVPR BASE&lt;/em&gt;. Congrats to first-author Dong!",description:"",section:"News"},{id:"news-jingyun-has-been-awarded-an-outstanding-ta-award-congrats-jingyun",title:"Jingyun has been awarded an Outstanding TA award. Congrats Jingyun!",description:"",section:"News"},{id:"news-our-paper-on-lt-a-href-quot-https-www-pnas-org-doi-abs-10-1073-pnas-2411930122-quot-gt-contextualized-networks-lt-a-gt-is-published-in-lt-em-gt-pnas-lt-em-gt-congrats-to-first-author-caleb",title:"Our paper on &lt;a href=&quot;https://www.pnas.org/doi/abs/10.1073/pnas.2411930122&quot;&gt;Contextualized Networks&lt;/a&gt; is published in &lt;em&gt;PNAS&lt;/em&gt;. Congrats to first-author Caleb!",description:"",section:"News"},{id:"news-prof-lengerich-appeared-on-lt-a-href-quot-https-www-youtube-com-watch-v-4z0ge0rwjrg-quot-gt-tomorrow-s-medicine-lt-a-gt-podcast",title:"Prof. Lengerich appeared on &lt;a href=&quot;https://www.youtube.com/watch?v=4z0ge0rwjrg&quot;&gt;Tomorrow\u2019s Medicine&lt;/a&gt; podcast.",description:"",section:"News"},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-3",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 3)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-3/"}},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-2",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 2)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-2/"}},{id:"posts-a-patient-s-perspective-on-the-future-of-medical-ai-part-1",title:"A Patient\u2019s Perspective on the Future of Medical AI (Part 1)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/a-patients-perspective-on-the-future-of-medical-ai-part-1/"}},{id:"posts-thoughts-from-my-faculty-application-experience",title:"Thoughts from my Faculty Application Experience",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2024/thoughts-from-my-faculty-application-experience/"}},{id:"posts-inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and",title:"Inverse Bradford Hill Criteria: How Association Flirts with Causality in Real-World Evidence, and\u2026",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/inverse-bradford-hill-criteria-how-association-flirts-with-causality-in-real-world-evidence-and/"}},{id:"posts-employment-numbers-adp-nfp-outliers-and-context-specific-correlation",title:"Employment Numbers: ADP, NFP, Outliers, and Context-Specific Correlation",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/employment-numbers-adp-nfp-outliers-and-context-specific-correlation/"}},{id:"posts-a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk",title:"A Paradox in Hospitalized Patient Outcomes: Why Do High-Risk Patients Do Better than Low-Risk?",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/a-paradox-in-hospitalized-patient-outcomes-why-do-high-risk-patients-do-better-than-low-risk/"}},{id:"posts-why-warren-buffett-doesn-t-invest-in-tech-companies-and-yet-his-largest-holding-is-apple",title:"Why Warren Buffett Doesn\u2019t Invest in Tech Companies (And yet his largest holding is Apple)",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-warren-buffett-doesnt-invest-in-tech-companies-and-yet-his-largest-holding-is-apple/"}},{id:"posts-why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects",title:"Why Inflation Chatter Is So Confusing: The Challenges of Additive Effects",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/why-inflation-chatter-is-so-confusing-the-challenges-of-additive-effects/"}},{id:"posts-the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games",title:"The Power of the Crowd: How Fan Attendance Impacts Hockey Games",description:"",section:"Posts",handler:()=>{window.location.href="/blog/2023/the-power-of-the-crowd-how-fan-attendance-impacts-hockey-games/"}},{id:"projects-contextualized-learning",title:"Contextualized learning",description:"Models with parameters that adapt to patient, task, or environmental context to support personalized inference.",section:"Projects",handler:()=>{window.location.href="/projects/1_contextualized/"}},{id:"projects-foundation-models-as-context",title:"Foundation models as context",description:"Using foundation models as flexible priors and contextual memory for downstream inference tasks.",section:"Projects",handler:()=>{window.location.href="/projects/2_foundation_models/"}},{id:"projects-foundations-of-intepretability",title:"Foundations of Intepretability",description:"Developing formal understanding of interpretability through structured models, GAMs, and interaction effects.",section:"Projects",handler:()=>{window.location.href="/projects/3_interpretable/"}},{id:"projects-efficient-foundation-models",title:"Efficient Foundation Models",description:"Structuring context improves inference within foundation models, just as it does in classical statistical models.",section:"Projects",handler:()=>{window.location.href="/projects/4_data_science/"}},{id:"projects-personalized-genomics",title:"Personalized Genomics",description:"Dissecting the biological bases of disease",section:"Projects",handler:()=>{window.location.href="/projects/5_genomics/"}},{id:"projects-clinical-ai-tools",title:"Clinical AI Tools",description:"Building tools for practicing clinicians",section:"Projects",handler:()=>{window.location.href="/projects/6_clinical_ai/"}},{id:"projects-real-world-evidence",title:"Real-world evidence",description:"Building AI systems to reveal surprises in observational EHRs",section:"Projects",handler:()=>{window.location.href="/projects/7_rwe/"}},{id:"projects-maternal-health",title:"Maternal health",description:"Applications to understand risk factors underlying maternal morbidity.",section:"Projects",handler:()=>{window.location.href="/projects/8_smm/"}},{id:"socials-email",title:"Send email",section:"Socials",handler:()=>{window.open("mailto:%6C%65%6E%67%65%72%69%63%68@%77%69%73%63.%65%64%75","_blank")}},{id:"socials-google-scholar",title:"Google Scholar",section:"Socials",handler:()=>{window.open("https://scholar.google.com/citations?user=a1Ck1CMAAAAJ","_blank")}},{id:"socials-github",title:"GitHub",section:"Socials",handler:()=>{window.open("https://github.com/AdaptInfer","_blank")}},{id:"socials-x",title:"X",description:"Twitter",section:"Socials",handler:()=>{window.open("https://twitter.com/AdaptInferLab","_blank")}},{id:"socials-rss",title:"RSS Feed",section:"Socials",handler:()=>{window.open("/feed.xml","_blank")}},{id:"light-theme",title:"Change theme to light",description:"Change the theme of the site to Light",section:"Theme",handler:()=>{setThemeSetting("light")}},{id:"dark-theme",title:"Change theme to dark",description:"Change the theme of the site to Dark",section:"Theme",handler:()=>{setThemeSetting("dark")}},{id:"system-theme",title:"Use system default theme",description:"Change the theme of the site to System Default",section:"Theme",handler:()=>{setThemeSetting("system")}}];</script> </body> </html>